### PR TITLE
dub: 1.33.0 -> 1.38.0

### PR DESCRIPTION
**Pull request by jtbx**
## Description of changes

https://github.com/dlang/dub/releases/tag/v1.34.0
https://github.com/dlang/dub/releases/tag/v1.35.0
https://github.com/dlang/dub/releases/tag/v1.36.0
**(edit 2024/06/06)**
https://github.com/dlang/dub/releases/tag/v1.37.0
https://github.com/dlang/dub/releases/tag/v1.38.0

Also migrate to pkgs/by-name and clean up the package.

Tested with nixpkgs-review.
```
3 packages built:
Literate btdu dub
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
@jtbx can you rebase to resolve conflicts ?

image

**Comment by jtbx** [comment]
Done

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 293624` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>Literate</li>
    <li>btdu</li>
    <li>dub</li>
  </ul>
</details>


here is a build log

[dub-1.36.0.log](https://github.com/NixOS/nixpkgs/files/14656425/dub-1.36.0.log)

**Comment by jtbx** [comment]
Weird, I encountered and fixed that error when I was writing this update. Running nixpkgs-review over this PR (on x86_64-linux) says that the build for all three succeeded?

**Comment by kirillrdy** [comment]
ofborg has same error

try rebuild locally ?

**Comment by jtbx** [comment]
No difference, same commit hashes and everything. I tried building on another machine running NixOS and I get your error, I'll look at this later thanks.

**Comment by jtbx** [comment]
Result of `nixpkgs-review pr 293624` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>Literate</li>
    <li>btdu</li>
    <li>dstep</li>
    <li>dub</li>
    <li>inochi-creator</li>
    <li>inochi-session</li>
    <li>luneta</li>
    <li>serve-d</li>
  </ul>
</details>

File: pkgs/by-name/du/dub/package.nix [hunk 1]
```diff
@@ -1,56 +1,45 @@
...
-        break
-      fi
-    done
-    if [ "$dc" == "" ]; then
-      exit "Error: could not find D compiler"
-    fi
-    echo "$dc_ found and used as D compiler to build $pname"
-    $dc ./build.d
-    ./build
+    runHook preBuild
```
Comment (kirillrdy) [inline pkgs/by-name/du/dub/package.nix:31]: nit pick 

usually there is a new line after runHook pre and new line before runHook post, but I'll leave it up to you

Comment (jtbx) [reply pkgs/by-name/du/dub/package.nix:31]: done, for all phases

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 293624` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>inochi-session</li>
  </ul>
</details>
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>Literate</li>
    <li>btdu</li>
    <li>dstep</li>
    <li>dub</li>
    <li>inochi-creator</li>
    <li>luneta</li>
    <li>serve-d</li>
  </ul>
</details>


on master dub builds for me on aarch64-linux

**Comment by jtbx** [comment]
Result of `nixpkgs-review pr 293624` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>inochi-session</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>Literate</li>
    <li>btdu</li>
    <li>dstep</li>
    <li>dub</li>
    <li>inochi-creator</li>
    <li>luneta</li>
    <li>serve-d</li>
  </ul>
</details>

**Comment by jtbx** [comment]
Everything builds fine for me, do you have a log for dub?

@ofborg build dub

**Comment by kirillrdy** [comment]
```
test/0-init-simple.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/1fzg4cl3k2n9yq80vg6y1vcmvx3qm682-bash-5.2p26/bin/bash"
test/issue1505-single-file-package-dynamic-library.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/1fzg4cl3k2n9yq80vg6y1vcmvx3qm682-bash-5.2p26/bin/bash"
test/issue1003-check-empty-ld-flags.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/1fzg4cl3k2n9yq80vg6y1vcmvx3qm682-bash-5.2p26/bin/bash"
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
Using pre-existing version file. To force a rebuild, provide an explicit version (first argument) or remove: /build/source/./source/dub/version_.d
No compiler has been found in the PATH. Attempted values: ["dmd", "ldmd2", "gdmd"]
Make sure one of those is in the PATH, or set the `DMD` variable
```

it's possible its because I am using binfmt,  if it works for you, good enough for me :-)

**Comment by jtbx** [comment]
Pushed a change that sets `DMD`, see if that works for you now

**Comment by kirillrdy** [comment]
interestingly different error

```
Using pre-existing version file. To force a rebuild, provide an explicit version (first argument) or remove: /build/source/./source/dub/version_.d
Building dub using /nix/store/34s81krabna0fm7wxxab6fgqfhg2ydmh-ldc-1.36.0/bin/ldmd2 (dflags: ["-g", "-O", "-w"]), this may take a while...
Command `["/nix/store/34s81krabna0fm7wxxab6fgqfhg2ydmh-ldc-1.36.0/bin/ldmd2", "-of/build/source/./bin/dub", "-I/build/source/./source", "-version=DubUseCurl", "-version=DubApplication", "-g", "-O", "-w", "@build-files.txt"]` failed, outp>

Error: /build/build-b04b81 failed with status: 1
```

no need to chase this for me, if it works for you that enough, i assume you tried on real aarch64 machine

**Comment by kirillrdy** [comment]
thank you !

**Pull request merged**